sdk/timeseriesinsights/Azure.Iot.TimeSeriesInsights/src/Models/TimeSeriesIdOfT1OfT2OfT3.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Iot.TimeSeriesInsights
+{
+    /// <summary>
+    /// A single Time Series Id value that is composed of three primitive values that uniquely identifies a Time Series instance.
+    /// </summary>
+    /// <remarks>
+    /// A single Time Series Id can be composite if multiple properties are specified as Time Series Id at environment creation time.
+    /// The position and type of values must match Time Series Id properties specified on the environment and returned by Get Model Setting API.
+    /// </remarks>
+    /// <typeparam name="T1">The type of the first Id property.</typeparam>
+    /// <typeparam name="T2">The type of the second Id property.</typeparam>
+    /// <typeparam name="T3">The type of the third Id property.</typeparam>
+    [System.Diagnostics.CodeAnalysis.SuppressMessage(
+        "StyleCop.CSharp.DocumentationRules",
+        "SA1649:File name should match first type name",
+        Justification = "The mismatch between the file name and the class name is because we have other TimeSeriesId classes that accept" +
+        "other generics. So in order to make it less confusing for the user, the class name has been simplified.")]
+    public class TimeSeriesId<T1, T2, T3> : ITimeSeriesId
```

Comment:
Did you not like Krzysztof's suggestion of having just once instance of the class with:
1. 3 ctors, one for each number used
2. 3 fields, tuples of whether it is used and the value

Reply:
For now, I changed the implementation based on the latest suggestion over the email thread. Will push that change to this PR and then we can discuss it further next week when we meet with Krzysztof.